src/evaluator/functions.rs

```diff
@@ -1089,6 +1089,69 @@ pub fn to_millis<'a>(
     }
 }
 
+pub fn fn_zip<'a>(
+    context: FunctionContext<'a, '_>,
+    args: &[&'a Value<'a>],
+) -> Result<&'a Value<'a>> {
+    if args
+        .iter()
+        .any(|arg| arg.is_null() || matches!(arg, Value::Undefined))
+    {
+        return Ok(context.arena.alloc(Value::Array(
+            bumpalo::vec![in context.arena;],
+            ArrayFlags::empty(),
+        )));
+    }
+
+    // Create a vector of arrays
+    let arrays: Vec<&bumpalo::collections::Vec<'a, &'a Value<'a>>> = args
+        .iter()
+        .filter_map(|arg| match arg {
+            Value::Array(arr, _) => Some(arr),
+            _ => None,
+        })
+        .collect();
+
+    if arrays.is_empty() {
+        let mut result: bumpalo::collections::Vec<&Value<'a>> = bumpalo::vec![in context.arena;];
+        let mut outer_array: bumpalo::collections::Vec<&Value<'a>> =
+            bumpalo::vec![in context.arena;];
+        for arg in args {
+            result.push(*arg);
+        }
+        outer_array.push(
+            context
```

Comment:
I think we can use the factory method to do the same thing. `Value::array_from(context.arena, result, ArrayFlags::empty())`